WordPress/gutenberg
# Site Editor: Improve loading experience (v1)

## What?
This is an experiment - DO NOT MERGE!

This PR aims to experiment with Suspense, specifically with our `useSuspenseSelect()` implementation to load the site editor behind the scenes, and display it once it's finished loading. 

## Why?

We're aiming to find a way to improve the editor loading experience.

As @mtias [says](https://github.com/WordPress/gutenberg/issues/35503#issuecomment-973182125):

> The end goal should be that things load like a "screenshot"; all fully loaded when you start interacting.

See #35503 for more details and motivation.

## How?

We're essentially replacing a bunch of `useSelect()` with `useSuspenseSelect`, adding a `Suspense` boundary with a dummy loading screen, and some artificial delay to prevent from flickering.

This is obviously not landable in this version - we'd likely need a better and more selective approach to which `useSelect` to alter and which not.

## Testing Instructions
* Load the site editor and observe the loading experience. 
* Observe that at the time the fallback placeholder has been hidden, the editor has fully loaded.

## Screenshots or screencast 

https://user-images.githubusercontent.com/8436925/179744565-eaa4deb3-6ffe-4185-b3c0-9c16b01a9973.mov
